extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalStreamCommandsImpl.java

```diff
@@ -219,4 +220,22 @@ public Uni<Void> xtrim(K key, XTrimArgs args) {
         this.tx.enqueue(Response::toLong);
         return this.reactive._xtrim(key, args).invoke(this::queuedOrDiscard).replaceWithVoid();
     }
+
+    @Override
+    public Uni<Void> xpending(K key, String group) {
+        this.tx.enqueue(reactive::decodeAsXPendingSummary);
+        return this.reactive._xpending(key, group).invoke(this::queuedOrDiscard).replaceWithVoid();
+    }
+
+    @Override
+    public Uni<Void> xpending(K key, String group, StreamRange range, int count) {
+        this.tx.enqueue(reactive::decodeListOfPendingMessages);
+        return this.reactive._xpending(key, group, range, count, null).invoke(this::queuedOrDiscard).replaceWithVoid();
+    }
+
+    @Override
+    public Uni<Void> xpending(K key, String group, StreamRange range, int count, XPendingArgs args) {
+        this.tx.enqueue(reactive::decodeAsXPendingSummary);
```

Comment:
should this be 
```suggestion
        this.tx.enqueue(reactive::decodeListOfPendingMessages);
```

Reply:
Oh good catch! 
My transaction test didn't use that method....